### ISSUE TITLE
stan_rdump writes NULL instead of zero-length objects

### ISSUE THREAD

**Issue by bgoodri** [stan-dev/rstan#366]
#### Summary:

It is not possible to round-trip lists of Stan data that have zero-length integers or doubles, although zero-length vectors and matrices work.

#### Description:

`stan_rdump` writes NULL instead of integer(), double(), etc.

#### Reproducible Steps:

```
library(rstan)
dat_list <- list(a = array(0, dim = c(3, 2, 1, 0)), d = double(), 
                     i = integer(), m = matrix(0, nrow = 3, ncol = 0))
stan_rdump(names(dat_list), file = "bug.data.R", envir = as.environment(dat_list))
str(read_rdump("bug.data.R"))
```

#### Current Output:

```
List of 4
 $ a: list()
  ..- attr(*, "dim")= int [1:4] 3 2 1 0
 $ d: NULL
 $ i: NULL
 $ m: list()
  ..- attr(*, "dim")= int [1:2] 3 0
```

#### Expected Output:

```
List of 4
 $ a: num[1:3, 1:2, 1, 0 ] 
 $ d: num(0) 
 $ i: int(0) 
 $ m: num[1:3, 0 ] 
```

#### RStan Version:

2.13.1

#### R Version:

3.3.2

#### Operating System:

Debian

**Comment by maverickg** [comment]
I will fix this. but stan_rdump will write integer(0) with dims if any in this cases for both integers and doubles.

**Comment by bgoodri** [comment]
Fixed by 370dcd7
